core/controllers/suggestion.py

```diff
@@ -344,8 +352,24 @@ def get(self, target_type, suggestion_type):
         """
         self._require_valid_suggestion_and_target_types(
             target_type, suggestion_type)
+        topic_name = self.request.get('topic_name', None)
+
+        opportunity_summary_exp_ids_specific_to_topic = []
+        if topic_name is not None:
+            topic = topic_fetchers.get_topic_by_name(topic_name)
+            if topic is not None:
```

Comment:
else raise exception?

Reply:
I am not really sure if we should raise an exception here. Ideally, the exception should be raised by function the "get_topic_by_name()" function in case a topic name is not found in the datastore so as to make sure that where-ever the function is called, the exception can be the same.

Reply:
I think I disagree; there may be cases where you want to try getting a topic by name to see whether there's a topic at all, and not want that to raise. But here it seems like we really don't expect the topic to be None.

Reply:
Yep, I agree, but just for my understanding, do you think we should fetch the complete topic in order to determine whether there's a valid topic at all? In those cases, we should probably create a new function called "is_topic_name_valid" which just returns a boolean (true/false) signifying whether a topic is valid or not. This will have two advantages: 

1. Clear functionality, i.e. get_topic_by_name gets the complete topic, whereas is_topic_name_valid returns whether a topic is valid or not. 
2. Anywhere when the get_topic_by_name is called in the entire code-base, we won't have to handle the case of getting None and throwing an exception and it can be done in a single place inside the function. 

This can be done in future, but just wanted to get your thoughts? In the meantime, I can throw an exception.

Reply:
I agree that is_topic_name_valid() sounds reasonable in general, yup.

But not so sure about the more general thing -- in general, other functions like get_exploration_by_id() reasonably return None and we deal with that appropriately, so it's kind of an established pattern at this point. I think the reason I commented here was because None isn't expected in this particular case, since we clearly expect a valid topic ID to be passed in (and in controllers we should properly handle all the cases).

Reply:
Done